### PR TITLE
Restart Index pod if does not have docker secret

### PR DESCRIPTION
**Pull request by mgencur**
Slack discussion: https://coreos.slack.com/archives/CD87JDUB0/p1652801062140119?thread_ts=1652789871.107389&cid=CD87JDUB0

### CODE REVIEW

File: hack/lib/catalogsource.bash [hunk 1]
```diff
@@ -79,6 +79,14 @@ spec:
   sourceType: grpc
 EOF
 
+  # Ensure the Index pod is created with the right pull secret. The Pod's service account needs
+  # to be linked with the right pull secret before creating the Pod. This is to prevent race conditions.
+  timeout 120 "[[ \$(oc -n $OLM_NAMESPACE get pods -l olm.catalogSource=serverless-operator --no-headers | wc -l) != 1 ]]"
+  index_pod=$(oc -n "$OLM_NAMESPACE" get pods -l olm.catalogSource=serverless-operator -oname)
+  if ! oc -n "$OLM_NAMESPACE" get "$index_pod" -ojsonpath='{.spec.imagePullSecrets}' | grep dockercfg &>/dev/null; then
+    oc -n "$OLM_NAMESPACE" delete pods -l olm.catalogSource=serverless-operator
```
Comment (mgencur) [inline hack/lib/catalogsource.bash:87]: Maybe I could put a little `sleep 5` before this line so as to make sure the ServiceAccount has the secret properly associate before restarting the Pod.

File: hack/lib/catalogsource.bash [hunk 2]
```diff
@@ -79,6 +79,15 @@ spec:
   sourceType: grpc
 EOF
 
+  # Ensure the Index pod is created with the right pull secret. The Pod's service account needs
+  # to be linked with the right pull secret before creating the Pod. This is to prevent race conditions.
+  timeout 120 "[[ \$(oc -n $OLM_NAMESPACE get pods -l olm.catalogSource=serverless-operator --no-headers | wc -l) != 1 ]]"
+  index_pod=$(oc -n "$OLM_NAMESPACE" get pods -l olm.catalogSource=serverless-operator -oname)
+  if ! oc -n "$OLM_NAMESPACE" get "$index_pod" -ojsonpath='{.spec.imagePullSecrets}' | grep dockercfg &>/dev/null; then
```
Comment (skonto) [inline hack/lib/catalogsource.bash:86]: get pods

Comment (mgencur) [reply hack/lib/catalogsource.bash:86]: No. The $index_pod is in the format `pod/serverless-operator-p9z6p` already.

Comment (skonto) [reply hack/lib/catalogsource.bash:86]: Ok

**Comment by mgencur** [comment]
The failure is not related. We can merge it as it is.

**Comment by skonto** [comment]
But we need that test no?

**Comment by mgencur** [comment]
> But we need that test no?

The test is optional. And it fails for a reason unrelated to this PR. IMO, it's totally fine to merge it.

**Comment by skonto** [comment]
ok let's merge. I will try again locally.

**Pull request merged**